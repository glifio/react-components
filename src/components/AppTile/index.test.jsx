import { render, screen } from '@testing-library/react'
import { Safe } from './index.stories'

describe('AppTile', () => {
  test('renders the AppTile', () => {
    const { container } = render(<Safe {...Safe.args} />)
    expect(screen.getByText(Safe.args.title)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
