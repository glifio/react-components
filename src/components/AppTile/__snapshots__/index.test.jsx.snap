// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTile renders the AppTile 1`] = `
.c0 {
  gap: var(--space-m);
}

.c1 {
  position: relative;
  font-size: var(--font-size-xl);
  line-height: 1.3em;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1 .hover {
  opacity: 0;
  -webkit-transform: scale(1.4);
  -ms-transform: scale(1.4);
  transform: scale(1.4);
}

.c4 {
  height: 100%;
  width: 100%;
  -o-object-fit: contain;
  object-fit: contain;
  position: absolute;
  top: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
  z-index: 2;
  font-size: var(--font-size-xl);
  line-height: 1;
}

.c2 {
  color: #fff;
  background: #000;
  border-radius: 10px;
  overflow: hidden;
}

.c5 {
  z-index: 1;
  font-size: var(--font-size-xl);
  line-height: 1.3;
}

.c5:hover {
  opacity: 1;
}

@media (max-width:868px) {
  .c0 {
    margin-top: var(--space-xl);
    margin-bottom: var(--space-xl);
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    overflow-x: scroll;
    -webkit-scrollbar-width: none;
    -moz-scrollbar-width: none;
    -ms-scrollbar-width: none;
    scrollbar-width: none;
    -ms-overflow-style: none;
    -ms-scroll-snap-type: x mandatory;
    -webkit-scroll-snap-type: x mandatory;
    -moz-scroll-snap-type: x mandatory;
    -ms-scroll-snap-type: x mandatory;
    scroll-snap-type: x mandatory;
  }

  .c0::-webkit-scrollbar {
    display: none;
  }
}

@media (min-width:868px) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:868px) {
  .c1 {
    width: 100%;
    overflow: hidden;
    padding-bottom: 144%;
  }
}

@media (max-width:868px) {
  .c1 {
    width: 45vw;
    -webkit-scroll-snap-align: start;
    -moz-scroll-snap-align: start;
    -ms-scroll-snap-align: start;
    scroll-snap-align: start;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
  }
}

@media (max-width:480px) {
  .c1 {
    width: 70vw;
    -webkit-scroll-snap-align: start;
    -moz-scroll-snap-align: start;
    -ms-scroll-snap-align: start;
    scroll-snap-align: start;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
  }
}

@media (hover:hover) {
  .c1:hover .hover {
    opacity: 1;
  }

  .c1:hover .default {
    opacity: 0;
  }
}

@media (hover:hover) {

}

@media (max-width:480px) {
  .c3 {
    padding: var(--space-m);
  }
}

@media (min-width:480px) {
  .c3 {
    padding: var(--space-l);
  }
}

@media (max-width:868px) {
  .c2 {
    height: 60vw;
    position: relative;
  }
}

@media (max-width:480px) {
  .c2 {
    height: 90vw;
    position: relative;
  }
}

@media (min-width:868px) {
  .c2 {
    position: absolute;
    width: 100%;
    height: 100%;
  }
}

@media (min-width:480px) and (hover:hover) {
  .c5 {
    opacity: 0;
    -webkit-transition: opacity 0.5s;
    transition: opacity 0.5s;
  }
}

@media (max-width:868px) {
  .c5 {
    width: 100%;
    white-space: normal;
    font-size: 1rem;
    padding: 15px 5px 0 var(--space-m);
  }
}

@media (min-width:868px) {
  .c5 {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    color: #fff;
    padding: var(--space-l);
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
    border-radius: 10px;
    opacity: 0;
  }
}

<div
  class="c0"
>
  <a
    class="c1"
    data-testid="app-tile-wrapper"
    href="/"
    rel="noreferrer"
    target="_blank"
  >
    <div
      class="c2"
    >
      <header
        class="c3"
      >
        <div>
          <h2
            style="margin: 0px; opacity: 1; text-decoration: none;"
          >
            Safe
          </h2>
          <h3
            style="opacity: 0.3; text-decoration: line-through; line-height: 1; margin: 0px;"
          >
            Vault
          </h3>
        </div>
      </header>
      <img
        alt=""
        class="c4 default"
        src="/static/bg-safe.jpg"
      />
      <img
        alt=""
        class="c4 hover"
        src="/static/bg-safe-hover.jpg"
        title="Safe"
      />
    </div>
    <div
      class="c5"
    >
      A Filecoin multisig wallet.
    </div>
  </a>
   
</div>
`;
